src/main/java/com/aws/greengrass/lifecyclemanager/Kernel.java

```diff
@@ -603,15 +612,64 @@ public Kernel parseArgs(String... args) {
         }
 
         // Update device configuration from commandline arguments after loading config files
-        kernelCommandLine.updateDeviceConfiguration(getContext().get(DeviceConfiguration.class));
+        DeviceConfiguration deviceConfiguration = getContext().get(DeviceConfiguration.class);
+        kernelCommandLine.updateDeviceConfiguration(deviceConfiguration);
+        // After configuration is fully loaded, initialize Nucleus service config
+        initializeNucleusServiceConfigIfNotExist(kernelAlts, deviceConfiguration);
+
         setupProxy();
+
         return this;
     }
 
     private void setupProxy() {
         ProxyUtils.setProxyProperties(context.get(DeviceConfiguration.class));
     }
 
+    private void initializeNucleusServiceConfigIfNotExist(KernelAlternatives kernelAlternatives,
+                                                          DeviceConfiguration deviceConfiguration) {
+        Topics nucleusLifecycle = getConfig().lookupTopics(DEFAULT_VALUE_TIMESTAMP, SERVICES_NAMESPACE_TOPIC,
+                deviceConfiguration.getNucleusComponentName(), SERVICE_LIFECYCLE_NAMESPACE_TOPIC);
+        if (nucleusLifecycle.children.size() != 0) {
+            logger.atDebug().log("Nucleus lifecycle has already been initialized");
+            return;
+        }
+
+        // Persist initial Nucleus launch parameters
```

Comment:
So we're moving this line in nucleus bootstrap script: `echo "{configuration:/jvmOptions}" > "$KERNEL_ROOT/alts/current/launch.params"`
to here. Nice!

Reply:
Yeah. The bootstrap script only gets to run in deployments. Here in provisioning I need to set it up explicitly